realm/realm-library/src/main/java/io/realm/RealmConfiguration.java

```diff
@@ -200,11 +200,25 @@ InputStream getAssetFile() throws IOException {
         return schemaMediator.getModelClasses();
     }
 
+    /**
+     * Returns the absolute path to where the Realm file will be saved.
+     *
+     * @return the absolute path to the Realm file defined by this configuration.
+     */
     public String getPath() {
         return canonicalPath;
     }
 
     /**
+     * Checks if the Realm file defined by this configuration already exists.
+     *
+     * @return {@code true} if the Realm file exists, {@code false} otherwise.
+     */
+    public boolean realmExists() {
```

Comment:
I kinda see your point, but I'm not entirely sure what kind of locks you think could make this better? This method has the exact same race condition as `new File(config.getPath()).exists()` which is what people would be using today?

But I'll just remove it from the public API for now.